Sources/MinderaPeople-iOS-DesignSystem/Components/Buttons/MinderaButton.swift

```diff
@@ -0,0 +1,195 @@
+import SwiftUI
+
+public struct MinderaButton: View {
+    // MARK: Inner Types
+
+    public enum Size {
+        case big
+        case small
+    }
+
+    public enum Style {
+        case primary
+        case secondary
+        case tertiary
+        case destructive
+        case danger
+    }
+
+    public enum ImagePosition {
+        case leading
+        case trailing
+    }
+
+    private struct ImageConfiguration {
+        let position: ImagePosition
+        let image: Image
+    }
+
+    // MARK: Private Properties
+
+    @Environment(\.colorScheme) private var colorScheme
+    private var size: Size = .big
+    private var style: Style = .primary
+    private var isDisabled = false
+    private var imageConfiguration: ImageConfiguration?
+
+    private let title: String?
+    private let action: () -> Void
+
+    // MARK: Lifecycle
+
+    public init(_ title: String?, action: @escaping () -> Void) {
```

Comment:
Suggestion: _Just to be more readable_
Instead checking the optional title (`let onlyIcon = title == nil`) why not adding a new enum for the Title / Type. Something like :
```
         case title(String)
         case noTitle
```
MinderaButton(.title("Hello world"), action: {})
MinderaButton(.noTitle, action: {})

Reply:
I saw that we also have ImageConfiguration as optional and IF's in `HStack` below
Another suggestion: Add more types
```
case titleAndImage(String, ImageConfiguration)
case image(ImageConfiguration) // case noTitle is not necessary anymore 
```
And remove the `func image(_ position: ImagePosition, image: Image` 
When we are creating the Button we are specifying the type of the button, if title, title and image or just image and with this you can remove the optionals
Also you can add var inside the enum that returns body for each type and then you just need to call 

(sorry I haven't touched SwiftUI in a long time so this is not good or even compile 😅)

```
label: {
          HStack(spacing: 8) {
                    type.body
....
```

Example
```
public enum Type:
		case title(String)
		case titleAndImage(String, ImageConfiguration)
		case image(ImageConfiguration)
		
		var body: some View {
			switch self {
			case .title:
				return titleBody
			case .titleAndImage(let title, let image):
				switch image.position {
			        case .leading:
					return
						imageBody(image)
						titleBody(title)
				case .trailing:
					return
					        titleBody(title)
						imageBody(image)
				}
			case .image(let image)
				return imageBody(image)
			}
		}
		
		private func titleBody(_ title: String): some View {
			Text(title)
				.customFont(size: .S, weight: .semiBold)
		}
		
		private var imageBody(_ image: Image): some View {
			image
				.resizable()
				.aspectRatio(contentMode: .fit)
				.frame(width: imageSize, height: imageSize)
		}
	}
```

This is also prepared for scalability if we need to add some new view or change order. 
Also because we have private funcs for each bodies if we need to change for example a property in the image we have just need to do it in one place

Reply:
done some similar approach. check it out